### DISCUSSION TITLE
Moving an ElementDocument programmatically

### DISCUSSION THREAD

**Discussion by barotto** [mikke89/RmlUi#243] [Q&A] [answered]
Hello,
I'm trying to solve an usability problem I have with my application.

I have a document that the user can move and resize.
This pose a problem when the application's client area shrinks and the document goes out of view (eg when going from fullscreen back to windowed)

So when that happens I move the document into view like so:
```
doc->SetOffset(Rml::Vector2f(new_x,new_y), nullptr);
```
So far so good, the document moves to the new position.
Unfortunately its position reverts back to the old values (and out of view again) as soon as:
1. the user tries to resize it
2. the user resizes the app's client area

What's the proper way to move a document?
Am I missing something?

**Comment by mikke89** [comment, marked as answer]
`SetOffset` is more of an internal function, I can certainly see how that is confusing. Instead, you should set the position of the document using RCSS properties, as the document is essentially just a normal, absolutely positioned element. In particular set the `top`, `left` properties to move the window, and `width`, `height` or `right`, `bottom` to size the window.

**Reply by barotto** [reply]
That works, but I decided to solve the problem in a less complicated way.
Suppose I have a document with this initial style in the rml:
```
body {
  width: 80%;
  height: 80%;
  margin: auto;
}
```
So size and position are auto updated when the app's window changes. But when the user resizes and moves the document, width/height and top/left will get fixed absolute values.

When the document goes out of view, I want to _reset_ to the initial values, as defined in the rml file.

Is there a way to do a reset? Or do I need to query the initial values when the document is loaded and then later use `SetProperty()`?

I tried to use `ReloadStyleSheet()` but it doesn't work.

**Reply by barotto** [reply]
Sorry, I just found the function I need: `RemoveProperty()`.
